packages/lib-classifier/src/components/Classifier/Classifier.spec.js

```diff
@@ -0,0 +1,507 @@
+import { Provider } from 'mobx-react'
+import { getSnapshot } from 'mobx-state-tree'
+import { Factory } from 'rosie'
+import sinon from 'sinon'
+
+import RootStore from '@store'
+import { ProjectFactory, SubjectFactory, TutorialFactory } from '@test/factories'
+import mockStore, { defaultAuthClient, defaultClient } from '@test/mockStore/mockStore'
+import branchingWorkflow from '@test/mockStore/branchingWorkflow'
+import Classifier from './Classifier'
+
+describe('Components > Classifier', function () {
+  this.timeout(0)
+
+  function withStore(store) {
+    return function Wrapper({ children }) {
+      return (
+        <Grommet theme={zooTheme}>
+          <Provider classifierStore={store}>
+            {children}
+          </Provider>
+        </Grommet>
+      )
+    }
+  }
+
+  describe('while the subject is loading', function () {
+    let subjectImage, tabPanel, taskAnswers, taskTab, tutorialTab, workflow
+
+    before(function () {
+      const subject = SubjectFactory.build({ locations: [{ 'image/png': 'https://foo.bar/example.png' }] })
+      const store = mockStore({ subject })
+      const project = store.projects.active
+      const projectSnapshot = { ...getSnapshot(project) }
+      workflow = store.workflows.active
+      const workflowSnapshot = { ...getSnapshot(workflow) }
+      render(
+        <Classifier
+          classifierStore={store}
+          project={projectSnapshot}
+          workflowID={workflowSnapshot?.id}
+          workflowSnapshot={workflowSnapshot}
+          workflowVersion={workflowSnapshot?.version}
+        />,
+        {
+          wrapper: withStore(store)
+        }
+      )
+      taskTab = screen.getByRole('tab', { name: 'TaskArea.task'})
+      tutorialTab = screen.getByRole('tab', { name: 'TaskArea.tutorial'})
+      subjectImage = screen.getByRole('img', { name: `Subject ${subject.id}` })
+      tabPanel = screen.getByRole('tabpanel', { name: '1 Tab Contents'})
+      const task = workflowSnapshot.tasks.T0
+      const getAnswerInput = answer => within(tabPanel).getByRole('radio', { name: answer.label })
+      taskAnswers = task.answers.map(getAnswerInput)
+    })
+
+    it('should have a task tab', function () {
+      expect(taskTab).to.be.ok()
+    })
+
+    it('should have a tutorial tab', function () {
+      expect(tutorialTab).to.be.ok()
+    })
+
+    it('should have a subject image', function () {
+      expect(subjectImage).to.be.ok()
+    })
+
+    describe('task answers', function () {
+      it('should be displayed', function () {
+        expect(taskAnswers).to.have.lengthOf(workflow.tasks.T0.answers.length)
+      })
+
+      it('should be linked to the task', function () {
+        taskAnswers.forEach(radioButton => {
+          expect(radioButton.name).to.equal('T0')
+        })
+      })
+
+      it('should be disabled', function () {
+        taskAnswers.forEach(radioButton => {
+          expect(radioButton.disabled).to.be.true()
+        })
+      })
```

Comment:
The added tutorial tests look good!

I want to go back to my previous point that many of Classifer's unit tests here check for the task tabs, subject image, and task answers, but those components belong to TaskArea. For example, whether the TaskComponent is disabled or not is controlled by Task and tested in its spec file:

https://github.com/zooniverse/front-end-monorepo/blob/880c355016987b40a74ac00fde336186a6c074a1/packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/Task/Task.spec.js#L104-L107

Classifier.js has a lot of `useEffect()` functions that change parts of the classifier store. I like your previous comment that this spec file can be expanded to test various types of subjects and workflows, and then we could observe the changes to the classifier store in each scenario. Disabled TaskArea components are already tested in their spec files, so I'm leaning toward Classifier unit tests focused only on the changes we expect in the classifier store - meaning "before and after a subject loads" blocks could be removed altogether. What do you think?

Reply:
The `TaskArea` tests don’t test the task alongside a subject viewer. That’s why we were able to deploy production code with a bug that passed the task and subject viewer tests, but showed a disabled task after the subject had loaded (#3250.) Tests here, with the appropriate subject and task, would have caught that bug and failed. 

The passing `TaskArea` tests are an example of the false positives discussed by Kent Dodds in his article about [testing implementation details](https://kentcdodds.com/blog/testing-implementation-details#false-positives).

Reply:
Re. The `useEffect` hooks, they update the store but the way to test store state, and still catch bugs, is to test the rendered HTML for a given store state. 

For example, if I click a link to navigate to a new workflow, the workflow ID change will fire an effect that updates the store, but I’d test that the subject image is the expected subject for that workflow, and the tasks shown are the expected tasks for that workflow. If my tests only check `store.workflows.active`, then I don’t have that confidence that the subject viewer and task area have updated to show the new workflow.

Reply:
That makes sense 👍